libnetdata/dictionary/dictionary.c

```diff
@@ -266,6 +289,58 @@ static int reference_counter_mark_deleted(DICTIONARY *dict, NAME_VALUE *nv) {
 // ----------------------------------------------------------------------------
 // hash table
 
+#ifdef DICTIONARY_WITH_AVL
+static int name_value_compare(void* a, void* b) {
+    if(((NAME_VALUE *)a)->hash < ((NAME_VALUE *)b)->hash) return -1;
+    else if(((NAME_VALUE *)a)->hash > ((NAME_VALUE *)b)->hash) return 1;
+    else return strcmp(((NAME_VALUE *)a)->name, ((NAME_VALUE *)b)->name);
+}
+
+static void hashtable_init_unsafe(DICTIONARY *dict) {
+    avl_init(&dict->values_index, name_value_compare);
+}
+
+static size_t hashtable_destroy_unsafe(DICTIONARY *dict) {
+    (void)dict;
+    return 0;
+}
+
+static inline int hashtable_delete_unsafe(DICTIONARY *dict, const char *name, size_t name_len, NAME_VALUE *nv) {
+    (void)name;
+    (void)name_len;
+
+    if(unlikely(avl_remove(&(dict->values_index), (avl_t *)(nv)) != (avl_t *)nv))
+        return 0;
+
+    return 1;
+}
+
+static inline NAME_VALUE *hashtable_get_unsafe(DICTIONARY *dict, const char *name, size_t name_len) {
+    (void)name_len;
+    uint32_t hash = simple_hash(name);
+
+    NAME_VALUE tmp;
+    tmp.hash = (hash)?hash:simple_hash(name);
+    tmp.name = (char *)name;
+    return (NAME_VALUE *)avl_search(&(dict->values_index), (avl_t *) &tmp);
+}
+
+static inline NAME_VALUE **hashtable_insert_unsafe(DICTIONARY *dict, const char *name, size_t name_len) {
```

Comment:
The function `hashtable_insert_unsafe` for avl is calling the previous function that does not have any call for [avl_insert](https://github.com/netdata/netdata/blob/3071aa055ccf9629278bd79aa2005291f5ad01c4/libnetdata/avl/avl.c#L43), instead it has a call for [avl_search](https://github.com/netdata/netdata/blob/3071aa055ccf9629278bd79aa2005291f5ad01c4/libnetdata/avl/avl.c#L20-L37) that moves trough all avl tree and returns a linked address without to insert new value.

On the other hand when we take a look in the `Judy` version we have a explicit call to `JudyHSIns` that inserts data.

There is an incompatibility between the `Judy` and `AVL` code, please, can you take a look on this?